### PR TITLE
Added Docs For filter and getindex in string

### PR DESCRIPTION
**Pull request by Akshat-mehrotra**

### CODE REVIEW

**Comment by Akshat-mehrotra** [comment]
What is going wrong? I've just forked the repo and added docs not changing the code at all. How could the tests fail!

**Comment by Akshat-mehrotra** [comment]
according to CONTRIBUTING.md,
I had to make some edits in the strings.md too. However, I am unsure that did I do it as required or not.

File: base/strings/string.jl [hunk 1]
```diff
@@ -206,6 +206,23 @@ function iterate_continued(s::String, i::Int, u::UInt32)
...
+
+# Examples
+
+```jldoctest
+julia> getindex("abcdef", 5)
+'e': ASCII/Unicode U+0065 (category Ll: Letter, lowercase)
+
+julia> getindex("Julia Is The Best Programming Language", 8)
+'s': ASCII/Unicode U+0073 (category Ll: Letter, lowercase)
+
```
Comment (logankilpatrick) [inline base/strings/string.jl]: I think this line needs to be removed.

File: base/strings/string.jl [hunk 2]
```diff
@@ -239,6 +256,25 @@ function getindex_continued(s::String, i::Int, u::UInt32)
     return reinterpret(Char, u)
 end
 
+"""
+    getindex(s::String, r::UnitRange{<:Integer})
+
+Returns a `String` derived from `s` which is situated at indexes from [`Base.first`](@ref) to [`Base.last`](@ref) in `s`.
+
+# Examples
+
```
Comment (logankilpatrick) [inline base/strings/string.jl]: remove this line

**Comment by Akshat-mehrotra** [comment]
Removed

**Comment by vtjnash** [comment]
Thank you @Akshat-mehrotra for the first-time contribution. However, I think this should be documented as the interface, such that these specific usages are merely implementations and don't require special mention. I've opened https://github.com/JuliaLang/julia/issues/40542 to discuss how we could improve the current printing to better clarify its role as an interface.

**Pull request closed without merging**